### PR TITLE
Simplifies the build steps

### PR DESCRIPTION
**Pull request by m-kuhn**
Simplifies the build steps to compile for Android to one single script

```
./scripts/build.sh
```

This script is also used on CI to create regular builds, so we can be sure it works.
This also integrates the git submodule into the source using git subtree instead. Removing another pitfall.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for cdda19851c58cb2f9aca68b1e4c3357a535e1e93:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-bc418177eae4076b853d60bfc688069ac211a3bd-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-bc418177eae4076b853d60bfc688069ac211a3bd-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-bc418177eae4076b853d60bfc688069ac211a3bd-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-bc418177eae4076b853d60bfc688069ac211a3bd-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for fe339cad659a4b3d9417b1ab774d3bd3b2b45e7a:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-585fcb13fba38689d306c1064f43b35b4580521d-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-585fcb13fba38689d306c1064f43b35b4580521d-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-585fcb13fba38689d306c1064f43b35b4580521d-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-585fcb13fba38689d306c1064f43b35b4580521d-x86.apk)

**Comment by m-kuhn** [comment]
@nirvn can you check this fixes https://github.com/opengisch/QField/issues/1538?
It runs fine here on Android 5 that shows a white screen with other 1.8 based builds.

**Comment by nirvn** [comment]
@m-kuhn , crash is still there.

**Comment by m-kuhn** [comment]
I assume you cleared cache and data once?

**Comment by nirvn** [comment]
I think so. The device is now out of battery, can't do more testing until tomorrow.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for e542b30ab32d3fa7fada760223dd33fdcd605374:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-40a70cec5b9ad32b0abf8936aa070a0c7b15a006-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-40a70cec5b9ad32b0abf8936aa070a0c7b15a006-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-40a70cec5b9ad32b0abf8936aa070a0c7b15a006-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-40a70cec5b9ad32b0abf8936aa070a0c7b15a006-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a9d9f9152976556b34d58932de3a69ccd7295024:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-f9d0d3131265ccb11eba09590744f39379d0f020-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-f9d0d3131265ccb11eba09590744f39379d0f020-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-f9d0d3131265ccb11eba09590744f39379d0f020-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-f9d0d3131265ccb11eba09590744f39379d0f020-x86.apk)

**Comment by m-kuhn** [comment]
This has been develipped further. And now removes all the hardcoded android versons do they are controlled through one single define in Osgeo4a.
And uses less tricks with rpath by using lld.
Anyone keen to review? Or trust and merge?

File: sdk.conf [hunk 1]
```diff
@@ -1 +1 @@
-osgeo4a_version=20201231
+osgeo4a_version=ndk_sideload
```
Comment (nirvn) [inline sdk.conf:1]: Did you want to update that to an official release tag before merging?

**Review by nirvn — APPROVED** [review]
I've tested an APK, runs fine.

**Comment by m-kuhn** [comment]
20210106 pipelined

**Comment by nirvn** [comment]
Should this be backported to 1.8 too? I think so.

**Comment by m-kuhn** [comment]
I don't see any direct benefits, so I wouldn't take the risk

**Comment by nirvn** [comment]
@m-kuhn , didn't you mention it fixed a blank screen for you?

**Comment by m-kuhn** [comment]
Not really in the end. It turned out that the screen just stayed white for a long time while "unpacking" asset data on first start. On older devices this is really not that nice ...

File: sdk.conf (hunk 1 above)
Comment (m-kuhn) [reply sdk.conf:1]:
```suggestion
osgeo4a_version=20210106
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 11bfbfb4bed0e35d70f16db6f701fb8120698619:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-6f9ae9eb75a8bb163fab0e760976dc89402cd0fa-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-6f9ae9eb75a8bb163fab0e760976dc89402cd0fa-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-6f9ae9eb75a8bb163fab0e760976dc89402cd0fa-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1558-6f9ae9eb75a8bb163fab0e760976dc89402cd0fa-x86.apk)

**Pull request merged**